hub/collections/dataset/__init__.py

```diff
@@ -132,3 +133,16 @@ def merge(datasets: Iterable[Dataset]) -> Dataset:
             assert tname not in tensors
             tensors[tname] = tvalue
     return Dataset(tensors)
+
+
+def from_tensorflow(ds, features, n_examples=-1):
```

Comment:
Great, this is more generalized! 
* Could you add dtags conversion
* Automatic compression selection based on dtag
* Then, support for large datasets when RAM is small and Numpy cannot handle (10GB+)